src/js/components/List/List.js

```diff
@@ -186,12 +186,14 @@ const List = React.forwardRef(
                   tabIndex: -1,
                   active: active === index,
                   onClick: event => {
-                    // extract from React's synthetic event pool
-                    event.persist();
-                    const adjustedEvent = event;
-                    adjustedEvent.item = item;
-                    adjustedEvent.index = index;
-                    onClickItem(adjustedEvent);
+                    if (event.type === 'click') {
```

Comment:
This looks a little awkward and slightly obscure, that the `onClick` would be called for an event other than `click`.

Would it produce the same effect if we just removed `onEnter` from the use of `Keyboard`? That feels a bit cleaner to me, if possible. As is, having the Keyboard onEnter here feels a bit like adding Keyboard with onEnter in a Form context and then preventing double submit by checking for the event.type on the submit button. Better just to not listen for Enter with Keyboard.

Reply:
I would agree with Eric's comment in terms of readability. Just to confirm, a "click" only occurs as an effect of mousedown/mouseup but not keyboard and that is why this specification solves the issue?

Reply:
@ericsoderberghp I agree that it feels awkward. It was feeling weird to me that the onClick was accepting other event types which was I opted to place the check there. I believe the ListItem `onClick` is using `<Box onClick />`.  I will check out your suggested approach.

@taysea - Confirming keyboard actions result in event types of `keydown`, `keypress`, and `keyup`. A mouse 'left click' results in event types of `mousedown`, `mouseup`, and `click`. A mouse 'right click' typically will fire an event type of `mousedown`, often opening a menu awaiting for other events to occur.

So yes, I was thinking of restricting the `onClick` to click events, but will check out alternatives.

Reply:
@ericsoderberghp Thanks for your question and suggested investigation. I modified approach to remove onEnter function, then needed to update onClick function to account for active item being set by keyboard interaction in addition to mouse events. This feels much better.

Note: I did need to slightly modify one Jest test to include a mouseOver event prior to the click event so that an active item had been set (models how a user would interact). Including the mouseOver did result in a snapshot change for that test because it changes the active state styling.

This should be ready for re-review.